### DISCUSSION TITLE
XUL links

### DISCUSSION THREAD

**Discussion by p1usminus** [p1usminus/roomybookmarksredux#29] [General]
Saving some of the things I found out about XUL here as many releases have been due to the removal of XUL in the browser...

## Removal ##
* https://groups.google.com/a/mozilla.org/g/firefox-dev/c/9sGpF1TNbLk/m/QpU3oTUuAgAJ
   * Mozilla dev thread on XUL layout removal progress which I found in January (IIRC.) As of March 2023 "all the XUL-box layout code in the tree is gone"
* https://crisal.io/words/2023/03/30/xul-layout-is-gone.html
   * This article summarises what was changed
* https://reddit.com/r/FirefoxCSS/comments/11odffm/psa_incoming_changes_to_default_element/?sort=new
   * Lists what was changed with the release of Firefox 113 (or 114 for Developer Edition)

## XUL notes ##
While trying to fix #27, I came across some very old XUL guides which brought up some interesting approaches.
* [How to resize a dialog using locale data](https://www.babelzilla.org/index.php?option=com_content&task=view&id=270&Itemid=)
* [Resize using sizeToContent on onLoad parameter](https://www.babelzilla.org/forum/index.php?showtopic=3229)
* [XUL reference archive](https://udn.realityripple.com/docs/Archive/Mozilla/XUL/XUL_Reference)
* [Dialog reference](https://docs.huihoo.com/mozilla/xul/elemref/ref_dialog.html), part of another larger XUL reference
* I discovered ondialog* was no longer supported during one of my [test commits](https://github.com/p1usminus/roomybookmarksredux/commit/341f490b335fbbc8a4572675412eb8891ed839eb), so I had to switch to the use of event listeners as in [this extension](https://github.com/cleidigh/EditEmailSubject-MX/commit/0530c9a119f822ec81f56cb6e7ab619c7f8609b8)

**Comment by 117649** [comment]
[XBL To Custom Element Converter](https://bgrins.github.io/xbl-analysis/converter/)
